### ISSUE TITLE
Feature Request: Add portion size to food

### ISSUE THREAD

**Issue by Cycor** [vkostyanetsky/ObsidianFoodiary#7]
I know it's a big change but would it be possible to add the portion in the product, so we can use portions as a descriptor. 
Also including unit type would make it clearer to know what the portion size means.
Products without portion could still be calculated the old way.

I would be a lot easier to just type:

Cheese 3
Eggs 1
Soup 200ml + 1
Donuts 85g

And the calories would then be per portion instead of 100g.

```
Cheese
---
calories: 103
fat: 8.5
portion: 28
unit: g
---

Soup
---
calories: 24
fat: 0.6
portion: 220
unit: ml
---

egg
----
calories: 72
fat: 4.8
potion: 50
unit: g

```

**Comment by Cycor** [comment]
Yes you are right about the point 1, the portion would be the default amount if no unit is given, so 1 would be 50g if potion: 50 and unit: g

the unit part would be used to make it clear what the portion means and also so we can add it when we want to use our own size instead of a portion.

For point 2, in the example 200ml includes ml so we don't use the default portion, the result would become 200 / portion * calories/fat/etc, and the 1 doesn't have a unit so that would mean the default portion, so 220ml in my example.

So that would become 200 + 220

**Comment by adamu** [comment]
Maybe a compromise to keep it simple could be to always require the products are specified in 100g, but also allow a portion size, which if supplied means the table expects number of portions, not weight.

So you could have sugar that's measured in grams, and apples that are measured in units:

```
Sugar
---
calories: 387
carbs: 100
---

Apple
---
calories: 52   // <- per 100g
carbs: 13.81   // <- per 100g
fat: 0.17      // <- per 100g
portion: 180   // <- 1 apple is 180g
---
```

Then we would enter it like this:

```
Sugar 100
Apple 1
```

A downside of this is that you would have to remember which system each product uses when entering the amounts. Maybe having a column in the results to remind which calculation is used could mitigate this. For example:


Product | Calories | Protein | Fat | Carbs | Method
-- | -- | -- | -- | -- | --
Sugar | 387 | 0 | 0 | 100 | ⚖️
Apple | 93.6 | 0 | 0.306 | 24.89 | 🧺
